firebase-common/src/main/java/com/google/firebase/FirebaseApp.java

```diff
@@ -536,7 +540,9 @@ protected FirebaseApp(Context applicationContext, String name, FirebaseOptions o
             registrars,
             Component.of(applicationContext, Context.class),
             Component.of(this, FirebaseApp.class),
-            Component.of(options, FirebaseOptions.class));
+            Component.of(options, FirebaseOptions.class),
+            SDKVersionComponentFactory.createComponent(FIREBASE_COMMON, BuildConfig.VERSION_NAME),
```

Comment:
optional nit: this seems a bit verbose and #enterprisy at the call site, would something like `SDKVersionComponent.create(String, String)` work?